server/server_test.go

```diff
@@ -91,6 +96,60 @@ func TestServerCall(t *testing.T) {
 			}
 		}
 	})
+	t.Run("Unimplemented hook", func(t *testing.T) {
+		t.Parallel()
+		var echoText = "are you there?"
+		var proxyReceived atomic.Value
+
+		// start a proxy server with hook
+		srv := server.New(nil, nil, nil)
+		srv.HandleUnknownMethod = func(method capnp.Method) *server.Method {
+			sm := server.Method{
+				Method: method,
+				Impl:   nil,
+			}
+			sm.Impl = func(ctx context.Context, call *server.Call) error {
+				echoArgs := air.Echo_echo_Params(call.Args())
+				inText, err := echoArgs.In()
+				require.NoError(t, err)
+				proxyReceived.Store(inText)
+				// pretend we received an answer
+				echo := air.Echo_echo{Call: call}
+				resp, _ := echo.AllocResults()
+				err = resp.SetOut(inText)
+				return err
+			}
+			return &sm
+		}
+		blankBoot := capnp.NewClient(srv)
+		lis, err := net.Listen("tcp", ":0")
```

Comment:
You can simplify this and entirely skip setting up a connection with something on the other side and just use `blankBoot` in place of `blankClient`

Reply:
@zenhack wouldn't that invoke 'Send' instead of 'Recv'?  The hook was intended for acting as a proxy forwarder and it doesn't exist in Send.

Reply:
O.O can't believe I missed this on my first look: Yeah, it would, and you really don't want the two to have different behavior -- the only difference that's supposed to be there is who allocates the arguments. Probably we should move the logic for method lookup into Server.start so it's not duplicated in both Send and Recv.

Reply:
Indeed it looks like there is some room for cleanup here as well. Combine methods.find, although that might not be as easy as it sounds.
{rant} Looking at the Send() function the return statement is rather confusing. I tried reading it a few times and am still not quite sure what is happening in what order. I've been bitten in the past by statements that do more  than one thing. In this case the return statement of Send() is akin to a rattlesnake den. The casual reader will benefit from having it split into a statement for each of the things it is doing. It looks like it needs the result of methods.find before srv.start is called but this is not quite clear to me. 

Would it be okay if  I add the hook to Send()  in the same way it is added to Recv and leave the cleanup to more capable folks?